joint_trajectory_controller/test/test_trajectory_controller.cpp

```diff
@@ -1233,8 +1238,41 @@ TEST_P(TrajectoryControllerTestParameterized, invalid_message)
   EXPECT_FALSE(traj_controller_->validate_trajectory_msg(traj_msg));
 }
 
-/// With allow_integration_in_goal_trajectories parameter trajectory missing position or
-/// velocities are accepted
+/**
+ * @brief invalid_message_nonzero_vel Test invalid velocity at trajectory end
+ */
+TEST_P(TrajectoryControllerTestParameterized, invalid_message_nonzero_vel)
+{
+  rclcpp::Parameter nonzero_vel_parameters("allow_nonzero_velocity_at_trajectory_end", false);
+  rclcpp::executors::SingleThreadedExecutor executor;
+  SetUpAndActivateTrajectoryController(executor, {nonzero_vel_parameters});
+
+  trajectory_msgs::msg::JointTrajectory traj_msg, good_traj_msg;
+
+  good_traj_msg.joint_names = joint_names_;
+  good_traj_msg.header.stamp = rclcpp::Time(0);
+  good_traj_msg.points.resize(1);
+  good_traj_msg.points[0].time_from_start = rclcpp::Duration::from_seconds(0.25);
+  good_traj_msg.points[0].positions.resize(1);
+  good_traj_msg.points[0].positions = {1.0, 2.0, 3.0};
+  good_traj_msg.points[0].velocities.resize(1);
+  good_traj_msg.points[0].velocities = {-1.0, -2.0, -3.0};
+  // Nonzero velocity at trajectory end!
+  EXPECT_FALSE(traj_controller_->validate_trajectory_msg(good_traj_msg));
```

Comment:
@christophfroehlich shouldn't this be True?, If I'm right, here you are parsing a valid trajectory message

Reply:
No, because this test is with `allow_nonzero_velocity_at_trajectory_end=false` -> the validate method should return false for this message.

Maybe I should rename the variable `good_traj_msg`, this was just a copy & paste.

Reply:
Ok, then I would totally recommend rename the variable and also a case where this is validated and returned true. I believe this second case is already tested along with others

Reply:
your are right, I renamed it now. 

There is another test right above where the same message with `allow_nonzero_velocity_at_trajectory_end=true` is checked for acceptance.